compat/src/main/scala-2.11_2.12/scala/collection/compat/CompatImpl.scala

```diff
@@ -12,10 +12,75 @@
 
 package scala.collection.compat
 
+import scala.collection.LinearSeq
 import scala.collection.generic.CanBuildFrom
 import scala.collection.mutable.Builder
 import scala.collection.{immutable => i, mutable => m}
 
+/* builder optimized for a single ++= call, which returns identity on result if possible
+ * and defers to the underlying builder if not.
+ */
+private final class IdentityPreservingSeqBuilder[A](that: Builder[A, Seq[A]])
```

Comment:
why not the following, so you can use the same class for both?

```scala
private final class IdentityPreservingSeqBuilder[A, CC[X] <: Seq[X]: ClassTag](that: Builder[A, CC[A]])
```

Reply:
That's a neat trick, I used it in https://github.com/scala/scala-collection-compat/pull/238/commits/ee6b3367c0e83497b0c7e80e0a7f95c8ccfda670

There is a lot of typecasting going on, which I'm not very experienced with, and I'm not sure this is the best way to do it, or the casts are neater elsewhere (e.g. on the inner builders) so please do review for correctness and style.